### DISCUSSION TITLE
Injected wallet appears twice

### DISCUSSION THREAD

**Discussion by alexboisseau** [wevm/wagmi#3395] [Q&A] [answered]
Versions : 
```json
"viem": "2.x",
"wagmi": "2.0.0-rc.3"
```

My wagmi configuration : 
```ts
import { createConfig, http } from "wagmi";
import { foundry, sepolia } from "wagmi/chains";
import { SUPPORTED_CHAINS } from "../constants/supported-chains";

const walletConnectProjectId = import.meta.env.VITE_WALLET_CONNECT_PROJECT_ID;
const foundryHttpTransport = import.meta.env.VITE_RPC_URL_FOUNDRY;
const sepoliaHttpTransport = import.meta.env.VITE_RPC_URL_SEPOLIA;

const config = createConfig({
  chains: SUPPORTED_CHAINS,
  connectors: [],
  transports: {
    [foundry.id]: http(foundryHttpTransport),
    [sepolia.id]: http(sepoliaHttpTransport),
  },
});

export { config, walletConnectProjectId };
```

Why, with this configuration, do I receive a connectors array with one item (Metamask)?
When I add `injected()` connectors from `wagmi/connectors`, I have two connectors (one for injected and another one for Metamask).

This is the metamask connector object that I have with 0 connectors in the configuration array : 
```json
{
    "id": "io.metamask",
    "name": "MetaMask",
    "type": "injected",
    "uid": "3e134989bff"
}
```

This is the injected connector object that I have with `injected()` in the configuration array (plus the Metamask connect which is above) : 
```json
{
    "id": "injected",
    "name": "Injected",
    "type": "injected",
    "uid": "e134989bff8"
}
```

**Comment by tmm** [comment, marked as answer]
In most cases, you shouldn't need to specify `injected` on your own. Wagmi automatically discovers injected providers via EIP-6963. You can disable this behavior by switching off [`multiInjectedProviderDiscovery`](https://wagmi.sh/react/api/createConfig#multiinjectedproviderdiscovery) if you really want.

**Reply by alexboisseau** [reply]
I have just found it in the documentation, sorry for the question ...
A big thanks for your answer and for your time !
